Source/JavaScriptCore/runtime/VM.cpp

```diff
@@ -1371,12 +1379,14 @@ ScratchBuffer* VM::scratchBufferForSize(size_t size)
     return result;
 }
 
-void VM::clearScratchBuffers()
+#if ASSERT_ENABLED
+void VM::assertScratchBuffersAreEmpty()
 {
     Locker locker { m_scratchBufferLock };
     for (auto* scratchBuffer : m_scratchBuffers)
-        scratchBuffer->setActiveLength(0);
+        ASSERT(!scratchBuffer->activeLength());
```

Comment:
Actually, we have a problem here. See https://trac.webkit.org/changeset/232490/webkit change's comment

Reply:
Thanks for catching this.  I restored it and trigger the clean up using an `EntryScopeService`.